test/testutils/util.go

```diff
@@ -128,6 +139,36 @@ func CreateLeaderPods(ctx context.Context, leaderSts appsv1.StatefulSet, k8sClie
 	if err != nil {
 		return err
 	}
+
+	return createLeaderPods(ctx, k8sClient, leaderSts, lws, podTemplateSpec, revisionutils.GetRevisionKey(cr), start, end)
+}
+
+func CreateLeaderPodsFromRevisionNumber(ctx context.Context, leaderSts appsv1.StatefulSet, k8sClient client.Client, lws *leaderworkerset.LeaderWorkerSet, start, end, revisionNumber int) {
+	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{MatchLabels: map[string]string{
+		leaderworkerset.SetNameLabelKey: lws.Name,
+	}})
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	revisions, err := revisionutils.ListRevisions(ctx, k8sClient, lws, selector)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	var oldRevision *appsv1.ControllerRevision
+	for i, revision := range revisions {
+		if revision.Revision == int64(revisionNumber) {
+			oldRevision = revisions[i]
+		}
+	}
+	oldLws, err := revisionutils.ApplyRevision(lws, oldRevision)
+	gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+	var podTemplateSpec corev1.PodTemplateSpec
+	// if leader template is nil, use worker template
+	if lws.Spec.LeaderWorkerTemplate.LeaderTemplate != nil {
+		podTemplateSpec = *oldLws.Spec.LeaderWorkerTemplate.LeaderTemplate.DeepCopy()
+	} else {
+		podTemplateSpec = *oldLws.Spec.LeaderWorkerTemplate.WorkerTemplate.DeepCopy()
+	}
```

Comment:
can we move this inside createLeaderPods? this will help remove one parameter from `createLeaderPods` func and dedup some logic that we also have in `CreateLeaderPods`